driven/storage/adapter_client.go

```diff
@@ -157,3 +207,57 @@ func (sa *Adapter) DeleteUserCourse(appID string, orgID string, userID string, c
 	}
 	return nil
 }
+
+// UpdateUserCourseStreaks updates streaks, pauses, and completed_tasks fieled
+func (sa *Adapter) UpdateUserCourseStreaks(appID string, orgID string, userID *string, userCourseID *string, courseKey string, streaks *int, pauses *int, userTime *time.Time) error {
```

Comment:
This storage function is for updating streaks, so streaks and pauses arguments should not be nullable. Also, since you are calling `UpdateOne`, userID should not be nullable because it is needed to uniquely identify a user course with appID, orgID, and courseKey.